Feat: 예약 내역 페이지 필터 기능 구현

miraclee1226 opened the pull request:
<!-- PR 제목은 커밋 메세지 컨벤션 형식으로 작성 -->

## 🧩 이슈 번호 <!-- 이슈 번호 입력 -->

- FP-56

## ✅ 작업 사항
예약 내역 페이지의 필터 기능과 예약 취소 기능 구현했습니다.

### 1. 필터 기능
예약 취소 내역 보여주는 스크린샷 첨부합니다. 다른 필터 기능들도(예약 확정, 예약 승인 등) 동일하게 동작합니다!
![스크린샷 2024-08-20 144110](https://github.com/user-attachments/assets/f3d544d0-92ef-4cc1-954a-c9149cf92ab0)


### 2. 예약 취소
![스크린샷 2024-08-20 144123](https://github.com/user-attachments/assets/1cf64549-7a2a-43d1-ab7f-2b2c9e787a4d)

## 👩‍💻 공유 포인트 및 논의 사항

miraclee1226 committed:
[FP-56] Feat: 예약 내역 페이지 API 함수 추가

miraclee1226 committed:
[FP-56] Feat: reservationId atom 추가

miraclee1226 committed:
[FP-56] Feat: 예약 내역 페이지 데이터 조회 기능 구현

miraclee1226 committed:
[FP-56] Refactor: ReservationCard 컴포넌트 개선 및 기능 추가

- MyReservation 타입을 사용하여 props 타입 변경
- reservationIdAtom을 사용하여 예약 ID 상태 관리 추가
- getButtonInfo 유틸리티 함수 도입으로 버튼 로직 개선

miraclee1226 committed:
[FP-56] Feat: 예약 내역 페이지 예약 내역 필터 기능 추가

miraclee1226 committed:
[FP-56] Chore:  CancelReservation 에서 CancelConfirm 컴포넌트로 교체

miraclee1226 committed:
[FP-56] Feat: 예약 내역 취소 기능 구현

miraclee1226 committed:
[FP-56] Refacotr: useActivitiesQuery 훅 제네릭 타입 추가 및 기능 리팩토링

miraclee1226 committed:
[FP-56] Fix: ActivitySettingsContent 컴포넌트 제목 텍스트 변경

miraclee1226 committed:
[FP-56] Fix: 빌드 에러 수정

tkddbs587 commented on src/components/common/Filter/index.tsx:
```diff
@@ -1,17 +1,32 @@
 import DownArrow from '@/../public/assets/icons/down-arrow.svg';
 import { useClickOutside } from '@/hooks/useClickOutside';
 import { useToggle } from '@/hooks/useToggle';
+import React from 'react';
```
P4) 이 친구는 생략 가능할 거 같습니다!

tkddbs587 commented on src/components/common/Modal/CancelConfirm/index.tsx:
```diff
@@ -1,22 +1,46 @@
+import { useAtomValue, useSetAtom } from 'jotai';
 import Image from 'next/image';
 
+import { usePatchMyReservationMutation } from '../../Cards/hooks/usePatchMyReservationMutation';
```
P5) 절대 경로로 통일해도 좋을 거 같아요!!

tkddbs587 commented on src/components/common/Modal/CancelConfirm/index.tsx:
```diff
@@ -1,22 +1,46 @@
+
 const CancelConfirmPopup = () => {
+  const setModalType = useSetAtom(modalAtom);
+  const activityId = useAtomValue(reservationIdAtom);
```
P4) 기본값은 reservationId인데 사용하는 값은 activityId라 헷갈릴 수 있을 거 같아서 통일 가능하시면 통일하는 것도 좋아 보이네요!

tkddbs587 commented on src/components/pages/myPage/ReservationList/reservationContent.tsx:
```diff
@@ -1,24 +1,39 @@
     >
       <h1 className="text-3xl-bold">예약 내역</h1>
-      {showFilter && !isEmpty && <Filter sortOptions={sortOptions} />}
+      {showFilter && !isEmpty && (
```
P4) 한번만 사용하는 조건부지만 코드가 조금 길어지더라도 변수 이름만 보고 어떤 조건부인지 명확하게 알 수 있도록 따로 변수화 시키는 것도 좋아보여요!!

tkddbs587 commented on src/components/common/Filter/index.tsx:
```diff
@@ -23,14 +38,20 @@ const Filter = ({ sortOptions }: FilterProps) => {
 
   const filterRef = useClickOutside(handleOutsideClick);
 
+  const getDisplayOption = (status: string | null) => {
+    if (!status) return content;
+
+    return Object.keys(statusMapping).find((key) => statusMapping[key] === status) || content;
+  };
```
P4) 이런 친구는 useMemo 등으로 메모이제이션 고려 해보셔도 좋을 거 같아요!

tkddbs587 reviewed (approved):
수고하셨습니다 혜린님!! 💪🏻

miraclee1226 committed:
[FP-56] Refactor: 변수명 변경

- activityId에서 reservation로 변경

miraclee1226 committed:
[FP-56] Chore: 불필요한 line 삭제

devjaesung reviewed (approved):
혜린님 수고하셨습니다!!

miraclee1226 committed:
[FP-56] Refactor: ReservationContent 컴포넌트 개선

- JSX 구조를 fragment(<>...</>)로 변경하여 불필요한 div 제거
- isEmpty prop을 isEmptyMyReservationData로 이름 변경하여 의미 명확화

miraclee1226 committed:
[FP-56] Refactor: ReservationList 컴포넌트 div 태그 추가

miraclee1226 committed:
Merge branch 'develop' of https://github.com/codeit-sprint6-team14/Global-Nomad into FP-56-feat/reservation-list-filter-api-connection

miraclee1226 committed:
[FP-56] Fix: 빌드 에러 해결

miraclee1226 committed:
[FP-56] Refactor: 언마운트시 쿼리 제거 기능 추가
